docs/apispec-docs/src/main/scala/sttp/tapir/docs/apispec/SecuritySchemesForEndpoints.scala

```diff
@@ -10,35 +10,38 @@ private[docs] object SecuritySchemesForEndpoints {
   def apply(es: Iterable[Endpoint[_, _, _, _]]): SecuritySchemes = {
     val auths = es.flatMap(e => e.input.auths)
     val authSecuritySchemes = auths.map(a => (a, authToSecurityScheme(a)))
-    val securitySchemes = authSecuritySchemes.map(_._2).toSet
-    val namedSecuritySchemes = nameSecuritySchemes(securitySchemes.toVector, Set(), Map())
+    val securitySchemes = authSecuritySchemes.map { case (auth, scheme) => auth.securitySchemeName -> scheme }.toSet
+    val takenNames = authSecuritySchemes.flatMap(_._1.securitySchemeName).toSet
+    val namedSecuritySchemes = nameSecuritySchemes(securitySchemes.toVector, takenNames, Map())
 
     authSecuritySchemes.map { case (a, s) => a -> ((namedSecuritySchemes(s), s)) }.toMap
   }
 
   @tailrec
   private def nameSecuritySchemes(
-      schemes: Vector[SecurityScheme],
+      schemes: Vector[(Option[String], SecurityScheme)],
```

Comment:
We could probably use the `SchemeName` alias here (which is just a string), but still the type `(Option[SchemeName], SecurityScheme)` would read better :)